otel.go

```diff
@@ -12,41 +13,50 @@ import (
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
+	otelTrace "go.opentelemetry.io/otel/trace"
+	"sync"
 )
 
 const (
-	packageName = "github.com/evergreen-ci/logkeeper"
+	packageName = "github.com/evergreen-ci/logkeeper/%s"
+	defaultName = "noop_tracer"
 )
 
-func (lk *logkeeper) initOtel(ctx context.Context) error {
-	if lk.opts.TraceCollectorEndpoint == "" {
-		// defaults to NoopTracerProvider
-		lk.tracer = otel.GetTracerProvider().Tracer(packageName)
-		return nil
-	}
+var (
+	closers       []closerOp
+	creator       sync.Once
+	closer        sync.Once
+	useCustomName bool
+)
 
+func LoadTraceProvider(ctx context.Context, useInsecure bool, collectorEndpoint string, sampleRatio float64) {
+	if collectorEndpoint == "" {
+		return
+	}
 	r, err := serviceResource(ctx)
 	if err != nil {
-		return errors.Wrap(err, "making host resource")
+		grip.EmergencyFatal(errors.Wrap(err, "making host resource"))
```

Comment:
maybe it could just return an error? Is this definitely something we'd want to kill the application over?